\

```diff
@@ -0,0 +1,245 @@
+# encoding: UTF-8
+require "logstash/devutils/rspec/spec_helper"
+require "logstash/outputs/file"
+require "logstash/event"
+require "logstash/json"
+require "stud/temporary"
+require "tempfile"
+require "uri"
+
+describe LogStash::Outputs::File do
+  describe "ship lots of events to a file" do
+    tmp_file = Tempfile.new('logstash-spec-output-file')
+    event_count = 10000 + rand(500)
+
+    config <<-CONFIG
+    input {
+      generator {
+        message => "hello world"
+        count => #{event_count}
+        type => "generator"
+      }
+    }
+    output {
+      file {
+        path => "#{tmp_file.path}"
+      }
+    }
+    CONFIG
+
+    agent do
+      line_num = 0
+      
+      # Now check all events for order and correctness.
+      tmp_file.each_line do |line|
+        event = LogStash::Event.new(LogStash::Json.load(line))
+        insist {event["message"]} == "hello world"
+        insist {event["sequence"]} == line_num
+        line_num += 1
+      end
+
+      insist {line_num} == event_count
+    end # agent
+  end
+
+  describe "ship lots of events to a file gzipped" do
+    Stud::Temporary.file('logstash-spec-output-file') do |tmp_file|
+      event_count = 10000 + rand(500)
+
+      config <<-CONFIG
+        input {
+          generator {
+            message => "hello world"
+            count => #{event_count}
+            type => "generator"
+          }
+        }
+        output {
+          file {
+            path => "#{tmp_file.path}"
+            gzip => true
+          }
+        }
+      CONFIG
+
+      agent do
+        line_num = 0
+        # Now check all events for order and correctness.
+        Zlib::GzipReader.open(tmp_file.path).each_line do |line|
+          event = LogStash::Event.new(LogStash::Json.load(line))
+          insist {event["message"]} == "hello world"
+          insist {event["sequence"]} == line_num
+          line_num += 1
+        end
+        insist {line_num} == event_count
+      end # agent
+    end
+  end
+
+  describe "#register" do
+
+
+    it 'doesnt allow the path to start with a dynamic string' do
+      path = '/%{name}'
+      output = LogStash::Outputs::File.new({ "path" => path })
```

Comment:
Didnt know that, you still have to declare path as a let variable to be able to override it.